src/poetry/installation/executor.py

```diff
@@ -600,7 +600,10 @@ def _install_git(self, operation: Install | Update) -> int:
 
         # Now we just need to install from the source directory
         original_url = package.source_url
-        package._source_url = str(source.path)
+        if package.source_subdirectory:
+            package._source_url = str(source.path / Path(package.source_subdirectory))
```

Comment:
I'm reasonably sure this extra `Path()` call is unnecessary -- what's the type of `package.source_subdirectory`?